peeknpoke.c

```diff
@@ -166,7 +166,7 @@ static int process_i2c_args(int argc, char **argv)
 		printf("\n***Invalid I2C operation\n");
 		status = -1;
 	}
-
+	free(values);
```

Comment:
Don't add free(values) here. Because malloc doesn't happen for every case. You should free values when values usage is done.